### PR TITLE
Extending feed-sel parameter syntax (#366)

### PR DESCRIPTION
**Pull request by zzeekk**
see #366

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/app/AppUtil.scala [hunk 1]
```diff
@@ -151,4 +152,58 @@ private[smartdatalake] object AppUtil extends SmartDataLakeLogger {
...
+   * - 'ids': select actions where id is matched by regex pattern
+   * - 'layers': select actions where metadata.layer of all output DataObjects is matched by regex pattern
+   * - 'startFromActionIds': select actions which with id is matched by regex pattern and any dependent action (=successors)
+   * - 'endWithActionIds': select actions which with id is matched by regex pattern and their predecessors
+   * - 'startFromDataObjectIds': select actions which have an input DataObject with id is matched by regex pattern and any dependent action (=successors)
+   * - 'endWithDataObjectIds': select actions which have an output DataObject with id is matched by regex pattern and their predecessors
+   * You can combine several pattern by using semicolon (;). Patterns are combined from left to right.
+   * By default combining sets is done with union operation. The operation can be modified by adding one of the following characters before the prefix:
+   * - pipe symbol (|): the two sets are combined by union operation (default)
+   * - ampersand symbol (&): the tow sets are combined by intersection operation
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/app/AppUtil.scala:170]:
```suggestion
   * - ampersand symbol (&): the two sets are combined by intersection operation
```

File: sdl-core/src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala [hunk 2]
```diff
@@ -120,13 +120,29 @@ abstract class SmartDataLakeBuilder extends SmartDataLakeLogger {
...
     head(appType, appVersion)
     opt[String]('f', "feed-sel")
       .required
       .action( (arg, config) => config.copy(feedSel = arg) )
-      .text("Regex pattern to select the feed to execute.")
+      .text("""Select actions to execute by one or multiple expressions separated by semicolon (;). Results from multiple expressions are combined from left to right.
+          |Expression syntax: "<operation?><prefix:?><regex>"
+          |Operations:
+          |- pipe symbol (|): the two sets are combined by union operation (default)
+          |- ampersand symbol (&): the tow sets are combined by intersection operation
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala:133]:
```suggestion
          |- ampersand symbol (&): the two sets are combined by intersection operation
```

File: sdl-core/src/main/scala/io/smartdatalake/util/misc/GraphUtil.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.util.misc
+
+private[smartdatalake] object GraphUtil {
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/util/misc/GraphUtil.scala:22]: Can you please add a short description of the object?

**Review by pgruetter — APPROVED** [review]
Minor scaladoc comments, otherwise LGTM

File: sdl-core/src/main/scala/io/smartdatalake/util/misc/GraphUtil.scala (hunk 3 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/util/misc/GraphUtil.scala:25]: Ok, done.

**Pull request merged**